### ISSUE TITLE
Tick gym

### ISSUE THREAD

**Issue by birgander2** [theCrag/website#2228]
I want to propose to add a kind of ticking feature for gyms. The result should be simply a stream event like "Some Name went climbing in gym ABC", without more details. Many gyms can be found on thecrag, but most are without entries (or outdated). Or I am to lasy or don't remember what to log exactly. However, a simple button to tick that I was climbing in that gym would be nice.

**Comment by scd** [comment]
Do you want to tick the gym node rather than individual routes?

**Comment by birgander2** [comment]
Yes, exactly. Because often the individual routes are not there.

**Comment by scd** [comment]
Is there any reason why this has to be restricted to gyms

**Comment by birgander2** [comment]
Well, not really, though much less important for real crags (in my opinion). But why not...

**Comment by brendanheywood** [comment]
This has a little bit in common with #2319 being able to 'Set' a route, or a whole cliff. Here we would want to 'climb' a whole cliff.

**Comment by quaestor** [comment]
+1. I wouldn't restrict it to gyms, it could just mean that you "visited" the crag/gym, even if you didn't climb there.

**Comment by brendanheywood** [comment]
This is essentially the same as a facebook check-in, you are stating you were there. You may not even have climbed anything. I've also wanted to do something to reflect scouting missions, or scrubbing / bolting sessions, see also some minor overlap with #1865 and #1217

**Comment by brendanheywood** [comment]
On reflection I think this does also make sense for an outside crag and even if you did end up climbing. It's kinda an edge case but often at new crags it would be nice if I could check in, and then I go find some people and have a chat and climb or whatever. When I go to log routes in the 'who' dropdown it would show me the list of recently climbed with people (https://github.com/theCrag/website/issues/2813), then friends, and then people with local karma, but it could also prioritize others who have checked in as a means of connecting with them. You would also be able to find them in the stream feed.

If somebody checks in to a crag I would treat it just like a ticking stream event, but initially with no ticks. If they don't tick anything then it remains like that, but then if they tick then it updates the existing event. This way you could checkin in the morning (which would probably be an automatic prompt in the app eventually), and then log ticks through out the day.

**Comment by brendanheywood** [comment]
On further reflection this also works well with the idea of merging photos, ticks and check-ins together. https://github.com/theCrag/website/issues/2791

**Comment by brendanheywood** [comment]
From @Drazhar via #3419 

Hey,

I couldn't find a similar issue, so I create a new one. Sorry if I missed something :)

It would be cool to be able to "tick" a crag/area. By tick I mean be able to tell that I visited that crag, give a comment and a rating. This would help to visit cool crags again on the next vacation or help friends find crags.

Most people I know also write comments in their guide books, this feature would represent this behavior.

Of cause one could argue that you also mark a visit by ticking routes and rate the area by the quality of the routes, but giving a comment to a area specifically is something different IMHO.

**Comment by Drazhar** [comment]
Thanks Brendan, it seems to be kinda related :)

My topic was more towards the ability to rate and comment crags (not with the "forum" feature), than to check in. It would be great to leave a personal opinion. Writing an description/approach/... is to much of a commitment for most people especially if you only visit a crag once.

**Comment by brendanheywood** [comment]
This is what I'm currently thinking:

1) you checking to a crag (or gym / comp)

2) you can optionally comment. At this point a 'checkin' is just a 'tick' stream event with no ticks. But I've been thinking it makes sense for tick events to have an optional generic comment. eg with strava you can hit record, go for a run, hit stop and post it and it gives you an automatic name like 'Afternoon run' which you can then edit. I think this works for most use cases like 'I went scouting', or 'Did a bunch of boulders but dunno what'. If you 'checkin' in facebook or most other apps you are prompted for a comment, and I think most people would intuitively understand the idea of what checkin means from other apps.

3) you can optionally rate it here, or perhaps rate it later on

4) If you do start ticking, it bundles up in the same event

There are two more things which I think would mesh conceptually:

5) Have a concept of auto-checking which you can opt in to. ie we have pretty good data for when you are 'in' a crag vs a region. So if you opened our app, and it says 'hey you are at Gara gorge, want to checkin?' it's one tap and you are done

6) It is very natural to want to check-in and either snap or upload some pics. I really think photos and ticks should be grouped together into the same stream events as almost every photo uploaded is a picture of someone climbing. See #2791. This is how facebook works, it is also how strava and other similar apps work.


The idea of people leaving a comment that is about the approach rather than updating the approach is interesting. I would prefer people to just improve things but it probably makes more sense when it is a bit vague or more of a question, and it prompts a local to 'answer' it by updating the real field. I think your tags comment was along the same lines, ie visitor: "went scouting with my kids, they loved it and I'll bring them next time with gear", local: "add kid friendly tag"

But a bit of balance is needed, I already see a lot of people starting a forum when there is a new access issue rather than just updating the access field. I want to make sure we steer people to adding data in the right places so it's not lost floating down the stream.

**Comment by Drazhar** [comment]
Yeah losing the data in the stream could be a problem. Because of this I thought these crag comments could be at a different place, for example below the images on the right.

**Comment by brendanheywood** [comment]
I wouldn't add them to the right hand column, we are already too cluttered and on mobile it will be worse. I think that would also give undue weight to these comments, they should be temporal in nature.

**Comment by georgkretzschmar** [comment]
> 
> 
> Well, not really, though much less important for real crags (in my opinion). But why not...

Actually here in Saxony there are lots of people who collect summits. Being able to tick summits (possibly automatically when you tick a route on a summit) would actually make a lot of sense and really increase the appeal of The Crag for Saxon and probably also Czech sandstone climbers. Furthermore this would hopefully also make it possible to make comments on a summit/crag/area.

**Comment by brendanheywood** [comment]
Interestingly if you tick, it creates the tick event group, and then if you remove the tick the event group remains. That is a bug, but it surprisingly worked pretty well exactly as how I would expect a check-in event to look, so this will be very little work when we get to it.
